node/services/jsonrpc/adminsvc/service.go

```diff
@@ -561,3 +571,45 @@ func (svc *Service) ResolutionStatus(ctx context.Context, req *adminjson.Resolut
 		},
 	}, nil
 }
+
+func (svc *Service) BlockExecStatus(ctx context.Context, req *adminjson.BlockExecStatusRequest) (*types.BlockExecutionStatus, *jsonrpc.Error) {
+	status := svc.app.BlockExecutionStatus()
+
+	if status == nil {
+		return nil, nil
+	}
+
+	resp := &types.BlockExecutionStatus{
+		Height:    status.Height,
+		StartTime: status.StartTime,
+		EndTime:   status.EndTime,
+	}
+	txInfo := make([]*types.TxInfo, len(status.TxIDs))
+	for i, txID := range status.TxIDs {
+		txInfo[i] = &types.TxInfo{
+			ID:     txID,
+			Status: status.TxStatus[txID.String()],
+		}
+	}
+	resp.TxInfo = txInfo
+	return resp, nil
+}
+
+func (svc *Service) RollbackBlock(ctx context.Context, req *adminjson.RollbackBlockRequest) (*adminjson.RollbackBlockResponse, *jsonrpc.Error) {
```

Comment:
I feel like this RPC is a big misleading.  You can't rollback any arbitrary block.  We were referring to it as "reset" or "abort", referring to an uncommitted consensus round?  Thoughts?

Reply:
That makes sense